airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -838,14 +839,23 @@ def primary_keys_for_records(streams, records):
     for stream in streams_with_primary_key:
         stream_records = [r for r in records if r.stream == stream.stream.name]
         for stream_record in stream_records:
-            pk_values = {}
-            for pk_path in stream.stream.source_defined_primary_key:
-                pk_value = reduce(lambda data, key: data.get(key) if isinstance(data, dict) else None, pk_path, stream_record.data)
-                pk_values[tuple(pk_path)] = pk_value
-
+            pk_values = _extract_primary_key(stream_record.data, stream.stream.source_defined_primary_key)
             yield pk_values, stream_record
 
 
+def _extract_primary_keys(records: Iterable[Mapping[str, Any]], primary_key: List[str]) -> Iterable[Tuple[Any, Mapping[str, Any]]]:
+    for record in records:
+        yield _extract_primary_key(record, primary_key)
+
+
+def _extract_primary_key(record: Mapping[str, Any], primary_key: List[str]):
```

Comment:
Can we unit test this function please to check if it works correctly with composite pk?

Reply:
There was indeed a bug there. Thanks for pointing that out